apps/contentful-apps/pages/fields/chart-component-source-data-key-field.tsx

```diff
@@ -0,0 +1,426 @@
+
+  const [textInput, setTextInput] = useState(externalSourceDataKey)
+
+  const [manualDataKeyValue, setManualDataKeyValue] = useState(
+    ManualDataKeyValues.Date,
+  )
+
+  const componentKeyValue = sdk.ids.entry
+
+  useEffect(() => {
+    sdk.window.startAutoResizer()
+  }, [sdk.window])
+
+  useDebounce(
+    () => {
+      sdk.entry.fields.values.setValue(manualData)
+    },
+    DEBOUNCE_TIME,
+    [manualData],
+  )
+
+  useEffect(() => {
+    setManualData((prevManualData) => ({
+      ...prevManualData,
+      typeOfSource: sourceDataKeyValue,
+      externalSourceDataKey: textInput,
+      ...(sourceDataKeyValue === SourceDataKeyValues.Manual && {
+        typeOfManualDataKey: manualDataKeyValue,
+      }),
+    }))
+  }, [sourceDataKeyValue, manualDataKeyValue, textInput])
+
+  useEffect(() => {
+    if (sourceDataKeyValue === SourceDataKeyValues.Manual) {
+      sdk.entry.fields.sourceDataKey.setValue(componentKeyValue)
+    } else {
+      sdk.entry.fields.sourceDataKey.setValue(textInput)
+    }
+  }, [
+    componentKeyValue,
+    sdk.entry.fields.sourceDataKey,
+    sourceDataKeyValue,
+    textInput,
+  ])
+
+  const handleTextInputChange = (event) => {
+    const newValue = event.target.value
+    setTextInput(newValue)
+  }
+
+  const updateManualDataDateItem = (
+    id: number,
+    key: keyof ManualData['dateItems'][number],
+    value: string,
+  ) => {
+    setManualData((prevManualData) => ({
+      ...prevManualData,
+      dateItems: prevManualData?.dateItems?.map((item) => {
+        if (item.id !== id) return item
+        return {
+          ...item,
+          [key]: value,
+        }
+      }),
+    }))
+  }
+  const updateManualDataCategoryItem = (
+    id: number,
+    key: keyof ManualData['categoryItems'][number],
+    value: string,
+  ) => {
+    setManualData((prevManualData) => ({
+      ...prevManualData,
+      categoryItems: prevManualData?.categoryItems?.map((item) => {
+        if (item.id !== id) return item
+        return {
+          ...item,
+          [key]: value,
+        }
+      }),
+    }))
+  }
+
+  const disableMoreThanOneItemOnPieChart =
+    sdk.entry.fields.type.getValue() === 'pie-cell' &&
+    (manualData?.categoryItems?.length > 0 || manualData?.dateItems?.length > 0)
+
+  return (
+    <>
+      <Stack flexDirection="row" marginBottom={'spacingM'}>
+        <Radio
+          id={SourceDataKeyValues.Manual}
+          name="manual"
+          value={SourceDataKeyValues.Manual}
+          isChecked={sourceDataKeyValue === 'manual'}
+          onChange={() => setSourceDataKeyValue(SourceDataKeyValues.Manual)}
+        >
+          Manual
+        </Radio>
+        <Radio
+          id={SourceDataKeyValues.ExternalSourceKey}
+          name="externalSourceKey"
+          value={SourceDataKeyValues.ExternalSourceKey}
+          isChecked={sourceDataKeyValue === 'externalSourceKey'}
+          onChange={() =>
+            setSourceDataKeyValue(SourceDataKeyValues.ExternalSourceKey)
```

Comment:
I think this value should be stored in the source data key field instead of inside the json